### ISSUE TITLE
Copyright violation

### ISSUE THREAD

**Issue by Poussinou** [martykan/forecastie#235]
Hi @avjinder

I found [this app](https://play.google.com/store/apps/details?id=com.rocasoftware.weatherapp) on the Google Play Store. This guy just took your source code, added some ads in it and put it on the Store, without warning this is free software...

As you are the owner of the source code, feel free to open a request to google [here](https://support.google.com/legal/contact/lr_dmca?product=googleplay) to remove the false app from the Store, if it's your wish ;)

Don't forget to tell Google that:

* The app is licensed under GNU GPL v3 license and this guy is violating it;
* He is falsely claiming he made the app;
* Also tell them that this guy has done the same thing already with other apps. Probably this is worth mentioning.

**Comment by bettina86** [comment]
I read the licence and you don't know what you say. I don't think that the gy is violating the copyright. Gnu licence v3 give you the possibility of modify, distribuite and sell the app and obviusly give other the possibility making the same thing. That guy should be write GNU licence v3 and give the link of source code.

**Comment by Poussinou** [comment]
@bettina86 YOU don't know "what you say". When publishing under GNU GPL v3, the guy who modify the app has to **publish the changes he made**, he **has to credit the original author**, and **has to add the license** or give a link to it... That's what I said "without warning this is free software" and "this guy is violating it"

Please read [the license](https://github.com/martykan/forecastie/blob/master/LICENSE) carefully one more time...

**Comment by FridoDeluxe** [comment]
@Poussinou's points are valid, thanks for discovering it and creating this issue! In my opinion this app should be reported. Maybe @martykan as the repository owner should do it? If he has no time I can do it, what does @robinpaulson think?

**Comment by robinpaulson** [comment]
At a quick glance it looks problematic. I will dig more, look for source code and attribution, then contact. Any help with a repo location would be appreciated.

**Comment by robinpaulson** [comment]
Also, please be careful when writing and reading messages folks; text is a dreadful medium for communication, meaning is often lost, confused or misunderstood.

**Comment by FridoDeluxe** [comment]
I couldn't find any source code or more information except in the app description at Google Play. The only website seems to be the [Privacy Policy](https://rocasoftware.blogspot.de/).

@robinpaulson You are right, it's probably the best to contact the developer first and ask for license/source code, give him maybe 7 days to respond and then report his app, if the issues aren't resolved.

**Comment by Poussinou** [comment]
@robinpaulson There is also [this one](https://play.google.com/store/apps/details?id=cz.martykan.forecastie)

Maybe you should take some time (one hour or something so) to check the other apps about weather information on Google Play Store, and report them to Google when this are just copies of Forecastie. I think some of them are just "Forecastie" with ads in it, and without GNU GPL, code source and your permission... If you do it, please copy-paste the links of the apps you reported in this thread, so that I can track the "Google developers" who are stealing your work. Thanks

**Comment by FridoDeluxe** [comment]
I found also [Weather forecast + widget](https://play.google.com/store/apps/details?id=com.vozisov.weather) and [Forecastnow](https://play.google.com/store/apps/details?id=com.ingic.forecastnow). Both look exactly as Forecastie, but do not mention a license or provide the source code.

Did anyone report the app from the first post? It's not available at Google Play Store anymore.

**Comment by Poussinou** [comment]
@FridoDeluxe Thanks for doing this
I do not report the first one, the fake developer is probably watching this thread... If you read this, fake developer, I'll take you down :heart:

**Comment by robinpaulson** [comment]
Interesting, good research @FridoDeluxe 
I haven't reported anything yet, it's on my todo list though.

**Comment by bettina86** [comment]
The developer of this app [https://play.google.com/store/apps/details?id=com.rocasoftware.weatherapp](https://play.google.com/store/apps/details?id=com.rocasoftware.weatherapp) answered me that he has the permission of Castica of use partially or full the code of Forecastie and he has a Commercial Licence. I don't know if it's true.

**Comment by martykan** [comment]
No, there is no commercial license. I told him he is authorized to
reproduce the app under the GPLv3 license - just like everyone else.

**Comment by FridoDeluxe** [comment]
The app https://play.google.com/store/apps/details?id=com.rocasoftware.weatherapp is available again at Google Play, but still neither licence nor source code I could find.

**Comment by robinpaulson** [comment]
When the software is installed, what does the "About" screen show? If it is a direct copy of Forecastie, there will be the author's name, a note about GPLv3 and a link to this github repo. I wonder if that is enough, or does GPLv3 explicitly require showing the licence and author _before_ installing? I would argue anyone _should_ show that info, but it may not be required.

That assumes they haven't made any modifications of course.

**Comment by FridoDeluxe** [comment]
My Forecastie 1.5 About screen contains the licences for weather data and artwork, and an email link to Tomas Martykan, but no repository URL or GPLv3 of the app itself.

I installed all three known potentially fake versions of Forecastie. All of them are not a exact copies but somewhat different:

[com.rocasoftware.weatherapp](https://play.google.com/store/apps/details?id=com.rocasoftware.weatherapp): 
- 500 - 1000 installations
- contains ads
- a few things seem to be changed or missing compared to Forecastie
- no About screen

[com.vozisov.weather](https://play.google.com/store/apps/details?id=com.vozisov.weather):
- 100 - 500 installations
- contains adds
- almost no changes
- About screen screenshot:
![weather forecast widget](https://user-images.githubusercontent.com/498537/29223028-54714838-7ec5-11e7-8440-5f7797d1835c.png)

https://play.google.com/store/apps/details?id=com.ingic.forecastnow
- 10 - 50 installations
- almost no changes
- no About screen

**Comment by robinpaulson** [comment]
@fridodeluxe perhaps we should add a repo link and see what happens next release?

**Comment by icasdri** [comment]
com.vozisov.weather is unquestionably in violation. Based solely on the screenshot given by @FridoDeluxe  above, it is a derivative work without any chance of full source code due to the inclusion of ads from ad networks which do not have open source licenses.

Furthermore, I would infer mal-intent from changing the author in the About, and therefore would be in favor of just filing DMCA takedown without any diplomacy. I will do this (I have code in Forecastie) if no one else wants to (although of course, @martykan doing it would be better).

**Comment by robinpaulson** [comment]
@icasdri I think you're right and there's little doubt this is unintentional. I have code in Forecastie too, go for it with the takedown. I think an attempt at diplomacy/discussion will likely go nowhere/waste time.

**Comment by FridoDeluxe** [comment]
Feel free to file a takedown request. But generally I'm personally in favor of the diplomatic way, in doubt the accused person might have just forgot to link the source code. Unfortunately a takedown request at a big company like Google starts a rather automatic and intransparent process that leaves almost no possibilities for the involed parties to react accordingly ([similar things happend to the Gadgetbridge project](https://blog.freeyourgadget.org/our-dmca-takedown-a-post-mortem.html))

Interestingly it's advantageous to have an F-Droid app published at Google Play, it makes things easier when handling fake copies: https://forum.f-droid.org/t/publish-your-gpl-apps-on-google-play-sooner-rather-than-later/827.

**Comment by icasdri** [comment]
@FridoDeluxe only problem with the case of com.vozisov.weather is that they can't possibly link to the source code because they are distributing a derivative work that contains closed-source ad software (and forecastie is GPL not LGPL). I will file DMCA for at least com.vozisov.weather (unless @martykan, who is better suited, would like to) when I find the time.

**Comment by robinpaulson** [comment]
@icasdri
Is it *necessarily* derivative, or is there any possibility it is mere aggregation?
I believe (but may be wrong) that if there are no calls between the two/reworking of forecastie code, it is not a derivative. For instance, could the ads be an overlay which ignores forecastie data/layout?

**Comment by icasdri** [comment]
@robinpaulson that would be the case if it were distributed separately. However, the code for ads and the app itself is bound together in one APK, and more specifically *linked* together (in the Java sense). Forecastie does not have the linking exception; therefore, under the GPL, I believe it does qualify as derivative (although IANAL).

**Comment by icasdri** [comment]
I think a basic understanding of the situation can be gleaned from https://www.gnu.org/licenses/gpl-faq.en.html#GPLIncompatibleLibs. 

*tl;dr*, if we had wanted the combination of Forecastie and non-free libs (in this case libraries providing ads) to be legally distributable, we would have had to explicitly grant an exception, which we have *not* done, and so they are not.

The only case where they may be is if the ads somehow qualify for the [system library exception](https://www.gnu.org/licenses/gpl-faq.en.html#SystemLibraryException), which is quite a stretch.

**Comment by robinpaulson** [comment]
> more specifically linked together (in the Java sense)

That's my point, is it *necessarily so* that they are linked? Is it possible they are not?

I agree if they are linked it is an obvious violation of GPL, my question is whether they are linked or not.

Yes, they are in one apk, but could that be seen as "mere combination" - after all, an APK is a zip archive and does not have to mean there are any calls between the included pieces of software? To be clear: I do not wish to defend gpl violations, merely to make sure we are on stable ground before we make accusations/do the DMCA thing.

**Comment by ildar** [comment]
Single classes.dex file → linked ?

**Comment by robinpaulson** [comment]
> Single classes.dex file → linked ?

OK, that's a good response. I don't understand the internals of a .dex file. I will research and find out.

**Comment by icasdri** [comment]
As I understand it, it can only be called an "aggregate" (defined in Section 5 of the GPL) only if one can claim that the ads and vanilla forecastie are *two separate programs*, as opposed to *one larger program*. (per Section 5: `A compilation ... with separate and independent works ... and which are not combined with it such as to form a larger program ... is called an "aggregate"`).

Supporting the spirit of this interpretation, a few paragraphs above, Section 5 also states `This License will therefore apply ... regardless of how they are packaged`.

**Comment by kabilzamandar** [comment]
I've built a weather forecast app, and came across Forecastie and i may use some part of the Forecastie code, i have my layouts, icons and other assets,  i need just few java classes. If/when i use some of the classes of the original code, does this mean i should open-source all of my app? I will distribute my app as  closed-source commercial app. I read GNU GPL v3 license but frankly i didn't get all of it. 

I also another question about licensed software,  if i use all of the code under GNU GPL v3 license and add it to my app or combine it my app, for example add a GNU GPL v3 license Exif reader app to my camera app, does it mean that i should distribute all of my code too? Where should i distribute the code, in GitHub? How can i commercially license my code?

**Comment by ildar** [comment]
1. You should read GPL FAQ at gnu.org
2. I'm almost sure libexif isn't GPL but LGPL

**Comment by FridoDeluxe** [comment]
@kabilzamandar On top of https://github.com/martykan/forecastie/blob/master/LICENSE is a small info box provided by GitHub that explains briefly what GPL means. Apart from that this issue is about the existing copyright violations and probably not the right place to discuss licencing plans.

**Comment by Poussinou** [comment]
Guys, there is another one [here](https://play.google.com/store/apps/details?id=com.mekayazilim.weatherforecast)...

**Comment by jawz101** [comment]
https://play.google.com/store/apps/details?id=com.alideveloper.weatherapps
https://play.google.com/store/apps/details?id=com.codebots.weather

**Comment by robinpaulson** [comment]
@martykan Are you willing to follow these up?

**Comment by FridoDeluxe** [comment]
https://play.google.com/store/apps/details?id=app.thienluan.forecastie

**Comment by FridoDeluxe** [comment]
I don't know how the clones handle the API key problem, probably they are using Forecastie's (unlimited) key?

**Comment by FridoDeluxe** [comment]
There are even amateurish copies without license on GitHub: https://github.com/NickHardwick2021/Forecastie.

**Comment by robinpaulson** [comment]
@FridoDeluxe Is there a way to report them to Github, for copyright infringement?

**Comment by ildar** [comment]
What's wrong with that besides false description?

**Comment by icasdri** [comment]
@ildar Uh... they literally deleted LICENSE (https://github.com/NickHardwick2021/Forecastie/commit/f6e1d3845ae6d123fc631af95ffd1a2e8907fedd) and are falsely claiming that it is their work. This not only violates the license and copyright notice requirement of the GPL but also is blatant plagiarism.

**Comment by ildar** [comment]
He didn't compile nor publish (distribute) it. What's the problem? The
crime isn't crime until it's done.
I suggest to concentrate on (now) distributing apps, not half-prepared.

**Comment by icasdri** [comment]
Publishing it on GitHub is conveying a verbatim (source) copy (see GPL section 4 "Conveying Verbatim Copies"), however in this case that conveying is being done without the required license and copyright notice. I do however agree that the infringing apps being distributed on the Play Store (especially the ones with ads) are more of a concern than this case.

**Comment by robinpaulson** [comment]
@icasdri takes the right position here @ildar . Github provides a way to access the code, i.e. it is published.

@martykan Could you take a look at this?

**Comment by robinpaulson** [comment]
Given that martykan is mostly busy, is anyone interested in pushing this? As a forecastie "collaborator", I have pretty much full control over the repository and releases, and am willing to grant permission to one or two of you who want to file DMCA takedown notices with github: https://support.github.com/contact/dmca-takedown

**Comment by robinpaulson** [comment]
Ditto Google Play, I assume they have a similar process.

**Comment by FridoDeluxe** [comment]
I prepared a mail text written from my perspective, a part of the text is based on a similar case at another Github repository. When I understand GPL section 8 correctly a GPL violator automatically regains the license if he fixes the problem in less than 30 days. That's why I inlcuded the 30 day deadline.

> Subject: GPL violation of $APP_NAME
> 
> Dear $ADDRESSEE,
> 
> You provide an Android app, "$APP_NAME", available via Google Play by the URL $GOOGLE_PLAY_URL. Your Android app has highly visible similarities in appearance as well as in code structure with "Forecastie" (available by the URL https://github.com/martykan/forecastie) which leads us to the assumption that Your Android app is a derivative work based on "Forecastie". "Forecastie" is licensed under the terms of GNU General Public License, version 3, as stated by the URL https://github.com/martykan/forecastie/blob/master/LICENSE.
> 
> As one of the contributors of "Forecastie" I would like You to either comply with the GNU General Public License, version 3, and provide a license notice and the complete source code for Your Android app or stop its distribution. Please take action within the next 30 days, otherwise we need to report a license violation to Google Play.
> 
> Sincerely,
> FridoDeluxe (on the behalf of "Forecastie" contributors)

Feel free to discuss and improve the text. If we send these mails still in November everything should be fixed by New Year or we can start filing takedown requests.

**Comment by FridoDeluxe** [comment]
@robinpaulson Any thoughts on the mail text?

**Comment by robinpaulson** [comment]
Yes, the text looks very good. Short, clear, to the point.

My only suggestion would be to make the language more forceful. So, replace "I would like you to" with "you must" and replace "please take action" with "I expect you to take action" or "you must take action". The law and generally-accepted moral values are on our side, I am in favour of showing off those principles unambiguously.

**Comment by FridoDeluxe** [comment]
Thanks for the feedback. The revised text (changes are highlighted):

> Subject: GPL violation of $APP_NAME
>
> Dear $ADDRESSEE,
>
> You provide an Android app, "$APP_NAME", available via Google Play by the URL $GOOGLE_PLAY_URL. Your Android app has highly visible similarities in appearance as well as in code structure with "Forecastie" (available by the URL https://github.com/martykan/forecastie) which leads us to the assumption that Your Android app is a modified work based on "Forecastie". "Forecastie" is licensed under the terms of the GNU General Public License, version 3, as stated by the URL https://github.com/martykan/forecastie/blob/master/LICENSE.
>
> As one of the contributors of "Forecastie" **I strongly insist on You either complying** with the GNU General Public License, version 3, and providing a license notice and the complete source code for Your Android app or stopping its distribution. **I expect you to take action** within the next 30 days, otherwise we need to report a license violation to Google Play.
>
> Sincerely,
> FridoDeluxe (on the behalf of "Forecastie" contributors)

I kept the phrase with "As one of the contributors of Forecastie" to make sure that I'm not just somebody unrelated to Foreastie.

**Comment by FridoDeluxe** [comment]
These are five GPL violating apps that are still available at Google Play. I checked there source code: Some are almost identical and even copied our OWM API key, others changed quite a lot and obfuscated the code, still it's obviously based on Forecastie. Looks like all of them include ads.

| Name | Application ID | Developer |
| --- | --- | --- |
| [Local Weather Live - Local Weather Forecast](https://play.google.com/store/apps/details?id=com.rocasoftware.weatherapp) | com.rocasoftware.weatherapp | Roca Software / Taba Apps |
| [The Weather](https://play.google.com/store/apps/details?id=com.alideveloper.weatherapps) | com.alideveloper.weatherapps | Ali Programmer |
| [Offline Weather Forecast - Maps & Radar](https://play.google.com/store/apps/details?id=com.codebots.weather) | com.codebots.weather | HazzelApps |
| [ForeCastie](https://play.google.com/store/apps/details?id=app.thienluan.forecastie) | app.thienluan.forecastie | Chim Viet |
| [Forecastie Weather](https://play.google.com/store/apps/details?id=vn.ntbltechh.wheatherappp) | vn.ntbltechh.wheatherappp | Android Apps Team |

@robinpaulson @martykan Is it okay if I send the above mail to the respective app developers? I will report back if they answer. If nothing is settled after 30 days I would file takedown requests at Google Play.

**Comment by robinpaulson** [comment]
Yes, this is great. Better, more forceful language. Please do contact the app publishers, then Google Play. Excellent work, I'm glad to see someone doing this.

**Comment by FridoDeluxe** [comment]
GPL proposes to add a copyright notice to every source code file. The notice should be someting like this:

> {one line to give the program's name and a brief idea of what it does.}
> Copyright (C) {year}  {name of author}
>
> This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
>
> This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
>
> You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.

Two questions arise for me:

1. Should we add such a copyright notice to every file? If yes, what is the name of the author: martykan or his real name, a name of every other contributor to the respective piece of code, or a link to the [contributors](https://github.com/martykan/forecastie/graphs/contributors)? I'm not sure if it's legally correct to add a copyright notice afterwards, and of course it means some work to go through all files.
2. If one of the GPL-violating Forecastie forks now aggrees to publish it under the GPL license how should the fork correctly credit the original Forecastie? Besides the license file and the text in the about screen there is currently nothing stating the license or the copyright.

**Comment by robinpaulson** [comment]
1. Yes, we should add copyright info to every file. Tomas Martykan is the author's real name, I think we should list it as "Tomas Martykan and other contributors",with a link to the Github page. Or we could look up how something like the Linux kernel does it, it's well known that Linus only contributes a tiny amount of code. I would suggest using the command-line utility sed to inject the applicable lines of code to all the files, a single line of bash will easily do that. I think it would be unreasonable for anyone to claim individual files are not GPL-covered, on the basis of us not including that line, I see it more as a convenience for someone who chances upon a single files without realising the broader context of it.
2. I'm not sure I understand. As far as I'm concerned, there has always been enough information for a person to realise the code was GPL and written by Tomas plus others, even if we did not fully follow the suggestions made by GNU. We are merely making that even more obvious now. Or have I missed the point you were making?

**Comment by FridoDeluxe** [comment]
1. Linux kernel uses [SPDX](https://spdx.org/) for license information in source code. I checked some other GPL projects, most of them don't use this copyright notice. But a copyright info with "Tomáš Martykán and other contributors" and a link to the Github contributors page sounds good. Also I fully agree on that:

> I think it would be unreasonable for anyone to claim individual files are not GPL-covered, on the basis of us not including that line, I see it more as a convenience for someone who chances upon a single files without realising the broader context of it.

2. My question was unclear. Besides the obvious GPL license I was wondering how exactly a fork needs to credit its origin in Forecastie: In GitHub's README file? In the description text at Google Play? In some kind of about screen inside the app? Or not at all, when there is no pre-existing copyright notice? For me the GPL does not have detailed instructions about this (probably for a good reason). But I guess something like "based on Forecastie" clearly visible in the app and its description should be enough.

**Comment by robinpaulson** [comment]
Right, got it. SPDX looks interesting.

I'm not sure how to answer your second point, but what you've written is probably a good answer. "Based upon Forecastie", with a link to the github repo sounds like enough for me.

**Comment by FridoDeluxe** [comment]
This is for tracking the status of the reported GPL violations and will be updated.

**com.rocasoftware.weatherapp**

- 2020-01-24 notice mail sent
- 2020-01-25 answer: _I would Gladly do it if you tell me exactly what i need to to to comply. I understand upload the sourdecode, How do i go about the GPL notice?_
- 2020-01-27 my reply: _Thank You for Your reply. Please understand that I cannot give You any legal advice. For the details about license compliance please refer to the terms of the GNU General Public License, version 3 (https://github.com/martykan/forecastie/blob/master/LICENSE), or the additional information given by the Free Software Foundation (https://www.gnu.org/licenses/gpl-3.0.en.html). In any case, for the case of "Forecastie" it is important that You: 1. license Your App "Local Weather Live - Local Weather Forecast" under the GNU General Public License, version 3 (and ensure that all code and libraries are compliant with this license), 2. provide the full source code, 3. state clearly that Your app is based on "Forecastie"._

**com.alideveloper.weatherapps**

- 2020-01-24 notice mail sent
- 2020-01-24 delivery fail: _mtaproxy312.free.mail.ne1.yahoo.com gave this error:
Remote server returned an error -> 554 delivery error: dd Sorry, your message to ***@yahoo.com cannot be delivered. This mailbox is disabled (554.30). - mta4226.mail.ne1.yahoo.com_
- 2020-01-28 DMCA takedown requested at Google

**com.codebots.weather**

- 2020-01-24 notice mail sent

**app.thienluan.forecastie**

- 2020-01-24 notice mail sent

**vn.ntbltechh.wheatherappp**

- 2020-01-24 notice mail sent

**Comment by FridoDeluxe** [comment]
Roca Software (the author of [com.rocasoftware.weatherapp](https://play.google.com/store/apps/details?id=com.rocasoftware.weatherapp)) is asking the following thing (the full conversation is in the post above):

> Im still not working on this as Im out of the City due to work. However i wanted to ask you and the team, Is there any way a can either get a paid license to use the code, or make an economical contribution to the development, to get a permission to use the app as it is? Ill be honest, the app does not generate real money with the ads, Im not gonna lie, its probably $5-7 a month. However I IF POSSIBLE i would prefer to leave it app as it is, 
> Let me know, Ill be taking a look on the app once I get back home. Ill try to make the mark you mention, Feb 24th. if any delay Ill let you know.
> Thanks for your support.

My opinion: He can join Forecastie development whenever he wants and he can use the code as long as he complies with the GPL license. But no exceptions, neither by any kind of payment.

@robinpaulson Right?

**Comment by robinpaulson** [comment]
@FridoDeluxe Absolutely, no exceptions.

**Comment by FridoDeluxe** [comment]
After almost two months all five GPL violating apps are not online anymore. One disappeared right after my contact attempt, the other four were made unavailable by Google. Other findings:
- A DMCA takedown request at Google takes some time. I asked for an update after a week and this seemed to accelerate the process.
- Some developers promised to fix license problems or asked for help, but they actually didn't change anything.

Feel free to use the texts above for future license violations. In case I handle any future license violations I will not report in such an extensive manner anymore.

**Comment by robinpaulson** [comment]
Marvellous, I'm glad to hear this. I very much appreciate your work following this up.

**Comment by Poussinou** [comment]
@FridoDeluxe Thanks for making everything clear, it was interesting to read!

**Comment by robinpaulson** [comment]
I've created a new issue to deal with SPDX info, take a look at #449 if you're interested.

**Comment by Prince655** [comment]
Hi sir, myself Prince. I'm working with this app since 1 month coz I found it very great. I have added many regional language in this as you know india is a agriculture land but our farmers are not well educated so regional language will make them understanding btw icons say much more. 
Another thing which I added is weather report that I'll manually update daily & of course ads a little bit.
So I want to ask you may I publish that on Play Store? 
If yes, then how coz I'm seeing you are taking down apps.
Can I publish it with the default about section to keep safe from takedown or may I add extra things in about me in about page. I have not any intention to take credit of your work, I'll only take credit what I did.
Pls reply,
Thanks.

**Comment by FridoDeluxe** [comment]
> ... So I want to ask you may I publish that on Play Store? ...

Forecastie is licensed under the GNU General Public License, version 3. That means you are allowed to use the source code and publish your modified app anytime as long as everything is compliant with this GPL license. Please inform yourself what's necessary for it. This issue's discussion has some information about the implications of using Forecastie's code.

Of course we would appreaciate if you could share your modifications with us in a way that Forecastie can benefit directly from these improvements. Feel free to discuss your ideas, send pull requests or help with translations.

**Comment by Prince655** [comment]
Ok sir, thanks.

**Comment by FridoDeluxe** [comment]
Mainly you need to
1. license your app under the GNU General Public License, version 3 (and ensure that all code and libraries are compliant with this license)
2. provide the full source code
3. state clearly that your app is based on "Forecastie"

**Comment by Prince655** [comment]
Yeah I'll give both source code and will clearly mention that this app is originally developed by "Thomas Martykan" Sir.
Thanks for simplify this is what I wanted to know.

**Comment by FridoDeluxe** [comment]
You also need to license your whole app under the GPL und ensure that all components are compliant. Please make sure that you understand what this means (we can't give you legal advice on your plans).

Please consider that it might be a more practiable way to contribute your favored modifications/improvements directly to Forecastie.

**Comment by Prince655** [comment]
I got it, I'll license my app under GPL 3 & also provide source code. Everyone will be able to fork and use that as it is (forecastie). 
I'll also do more research on it coz it's gonna very big project.

**Comment by robinpaulson** [comment]
> Yeah I'll give both source code and will clearly mention that this app is originally developed by "Thomas Martykan" Sir.
> Thanks for simplify this is what I wanted to know.

It's good to hear from you @Prince655 , and to know you are interested in Forecastie.

I will echo Frido's suggestion of considering submitting changes directly to our repo. The app currently supports several languages, I would be very happy to see that list extended.

Please be sure to check the name of the original author, it's "Tomáš Martykán".

Thanks

**Comment by Prince655** [comment]
You welcome, It was pretty good to submit with this repo but I just want to keep forecastie fresh and clean. 
Btw everyone will have access to all languages which I added: bangla, hindi, urdu, tamil, telugu. I will make everything public very soon and continuously make changes.

**Comment by hefty-fading-culprit-epiphany-drew** [comment]
Lol Foss police at work, btw this app is awesome, pity that I didn't discover it earlier, it is like the arch Linux of weather apps, simple to the point

**Comment by Poussinou** [comment]
Another one here https://play.google.com/store/apps/details?id=weather.reports.rb.com from [this developer](https://play.google.com/store/apps/dev?id=5161378688186728618).

**Comment by FridoDeluxe** [comment]
The screenshots look more than fishy. I installed this app out of curiousity and it doesn't work, only this is displayed:

> Temporarily Service Down, Sorry For Inconvinence

The contents of th decomplied `.apk` do not look familiar to me. Maybe only Forecastie based images were used for this app?

**Comment by Lee-Carre** [comment]
The [GPL Violations Project](//www.gpl-violations.org/) may be interested in this.
